### PR TITLE
[#1316] fix: Support keyword in column name in MySQL/PG catalog

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

back quoted keyword like `long`, 'integer' in column name

### Why are the changes needed?

Users may use this keyword to create table.

Fix: #1316 

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

IT

### CODE REVIEW

**Comment by jerryshao** [comment]
Do we also need to fix PostgreSQL?

**Comment by yuqi1129** [comment]
> Do we also need to fix PostgreSQL?

I have added the keyword check for PG, PG uses '"' to quote keyword.

**Comment by jerryshao** [comment]
@Clearvive please help to review this.

**Comment by jerryshao** [comment]
I think it is a good fix, we should avoid any corner case.

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/catalog/jdbc/postgresql/CatalogPostgreSqlIT.java [hunk 1]
```diff
@@ -195,6 +197,80 @@ private ColumnDTO[] createColumns() {
...
+  void testCreateTableWithSpecialColumnNames() {
+    // Create table from Gravitino API
+    ColumnDTO[] columns = columnsWithSpecialNames();
+
+    NameIdentifier tableIdentifier =
+        NameIdentifier.of(metalakeName, catalogName, schemaName, tableName);
+    Distribution distribution = Distributions.NONE;
+
+    SortOrder[] sortOrders = new SortOrder[0];
+    Partitioning[] partitioning = Partitioning.EMPTY_PARTITIONING;
```
Comment (jerryshao) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/catalog/jdbc/postgresql/CatalogPostgreSqlIT.java:255]: I think you should use `Transform[0]` instead of `Partitioning`. `Partitioning` should not be exposed to the user.

Comment (yuqi1129) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/catalog/jdbc/postgresql/CatalogPostgreSqlIT.java]: Addressed.

Comment (yuqi1129) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/catalog/jdbc/postgresql/CatalogPostgreSqlIT.java]: I have replaced `new Transform[0]` with `Transforms.EMPTY_TRANSFORM`.

**Review by jerryshao — APPROVED** [review]
